Why am I getting this dagger/anvil erro

```
/Users/hossain.khan/dev/repos/hk-repos/android-trmnl-display/app/build/generated/source/kapt/debug/dev/hossain/trmnl/di/DaggerAppComponent.java:165: error: incompatible types: Provider<TrmnlDisplayRepository> cannot be converted to Provider<TokenManager>
      this.trmnlMirrorDisplayPresenterProvider = TrmnlMirrorDisplayPresenter_Factory.create(trmnlDisplayRepositoryProvider, tokenManagerProvider, trmnlWorkManagerProvider, imageMetadataStoreProvider);
                                                                                            ^
```

and 

```
/Users/hossain.khan/dev/repos/hk-repos/android-trmnl-display/app/build/generated/source/kapt/debug/dev/hossain/trmnl/di/DaggerAppComponent.java:176: error: method create in class TrmnlImageUpdateManager_Factory cannot be applied to given types;
      this.trmnlImageUpdateManagerProvider = DoubleCheck.provider(TrmnlImageUpdateManager_Factory.create());
                                                                                                 ^
  required: Provider<ImageMetadataStore>
  found:    no arguments
  reason: actual and formal argument lists differ in length
```

--- a/app/src/main/java/dev/hossain/trmnl/data/ImageMetadataStore.kt
+++ b/app/src/main/java/dev/hossain/trmnl/data/ImageMetadataStore.kt
@@ -7,8 +7,6 @@ import androidx.datastore.preferences.core.edit
 import androidx.datastore.preferences.core.longPreferencesKey
 import androidx.datastore.preferences.core.stringPreferencesKey
 import androidx.datastore.preferences.preferencesDataStore
-import com.squareup.anvil.annotations.optional.SingleIn
-import dev.hossain.trmnl.di.AppScope
 import dev.hossain.trmnl.di.ApplicationContext
 import kotlinx.coroutines.flow.Flow
 import kotlinx.coroutines.flow.first
@@ -28,7 +26,6 @@ private val Context.imageDataStore: DataStore<Preferences> by preferencesDataSto
  *
  * @see ImageMetadata
  */
-@SingleIn(AppScope::class)
 class ImageMetadataStore
     @Inject
     constructor(

--- a/app/src/main/java/dev/hossain/trmnl/ui/display/TrmnlMirrorDisplayScreen.kt
+++ b/app/src/main/java/dev/hossain/trmnl/ui/display/TrmnlMirrorDisplayScreen.kt
@@ -105,13 +105,18 @@ class TrmnlMirrorDisplayPresenter
             var error by remember { mutableStateOf<String?>(null) }
             val scope = rememberCoroutineScope()
 
-            // Collect updates from the image update manager
+            // Collect updates from the image update manager to get the latest image URL
+            // Latest image URL is received from WorkManager work requests.
             LaunchedEffect(Unit) {
                 trmnlImageUpdateManager.imageUpdateFlow.collect { newImageUrl ->
                     if (newImageUrl != null) {
                         imageUrl = newImageUrl
                         isLoading = false
                         error = null
+                    } else {
+                        Timber.w("Failed to get cached image URL from WorkManager")
+                        isLoading = true
+                        error = "Failed to load image"
                     }
                 }
             }

